codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/transformers/EventStreamNormalizer.kt

```diff
@@ -37,3 +59,38 @@ object EventStreamNormalizer {
             .build()
     }
 }
+
+private val allErrors = mutableMapOf<String, OperationErrors>()
+@Synchronized fun computeErrors(model: Model, operationShape: OperationShape) {
+    val errors = allErrors.computeIfAbsent(operationShape.id.name) {
+        val getErrors = { shape: ShapeId ->
+            model.expectShape(shape).members()
+                .filter { it.isEventStream(model) }
+                .map { model.expectShape(it.target) }.flatMap { it.members() }
+                .map { model.expectShape(it.target) }
+                .filter { it.hasTrait<ErrorTrait>() }
+        }
+        val input = operationShape.input.map { getErrors(it) }
+        val output = operationShape.output.map { getErrors(it) }
+        val inputOutput = input.orElse(listOf()) + output.orElse(listOf())
+        val operationIndex = OperationIndex.of(model)
+        OperationErrors(
+            operationIndex.getErrors(operationShape).filter { !inputOutput.contains(it) },
+            inputOutput.filter { it.getTrait<ErrorTrait>()!!.isServerError },
+            inputOutput.filter { it.getTrait<ErrorTrait>()!!.isClientError },
+        )
+    }
+    allErrors[operationShape.id.name] = errors
+}
+fun OperationShape.getStreamError(target: CodegenTarget): List<Shape> {
+    return allErrors[this.id.name]!!.getStreamError(target)
+}
+fun OperationShape.nonEventStreamErrors(): List<Shape> {
```

Comment:
I think we should call them "operation errors" and "event stream errors" rather than "non event stream errors".